data/json/portal_traveler_effect_on_condition.json

```
@@ -0,0 +1,467 @@
+      { "u_message": "You suddenly feel as if invisible shackles were being loosened in your ankles.", "type": "good" },
+      {
+        "u_add_morale": "morale_feeling_good",
+        "bonus": 20,
+        "max_bonus": 40,
+        "duration": "60 minutes",
+        "decay_start": "60 minutes"
+      }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_PORTAL_TRAVELER_ACCEPTANCE_4",
+    "//": "Your feel greatly more accepted by the world, your body partially stabilizes itself in this reality if you are outside.",
+    "effect": {
+      "set_queue_effect_on_condition": [
+        {
+          "id": "EOC_PORTAL_TRAVELER_ACCEPTANCE_OUTSIDE",
+          "condition": "u_is_outside",
+          "effect": [
+            {
+              "u_message": "You suddenly feel a warm embrace by your surroundings, it does not lasts long, but you now feel a weak connection with the world around you.",
+              "type": "good"
+            },
+            { "u_add_trait": "REALITY_GRASP" }
+          ],
+          "false_effect": {
+            "u_message": "You suddenly feel a warm embrace by your surroundings, you feel like something is about to change inside you but the embrace seems to lack strenght and stops just before you could grasp it… You now feel a strong need to go out and jump into the portal storm.",
+            "type": "warning"
+          }
+        }
+      ]
+    }
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_PORTAL_TRAVELER_FOCUS_GOOD",
+    "//": "Your mind gains focus as an effect of your connection with the portal storms.",
+    "condition": { "not": { "u_has_effect": "sleep" } },
+    "effect": [
+      { "u_message": "Everything suddenly feels more clear now.", "type": "good" },
+      { "arithmetic": [ { "u_val": "focus" }, "+=", { "const": 10 } ] }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_PORTAL_TRAVELER_MESSAGES_GOOD",
+    "//": "Messages about good things that you hear/see related to your condition.",
+    "condition": { "and": [ { "not": { "u_has_effect": "sleep" } }, "u_can_see" ] },
+    "effect": [ { "u_message": "PORTAL_TRAVELER_MESSAGES_VARIED_GOOD", "snippet": true, "type": "mixed" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_PORTAL_TRAVELER_EFFECTS_REJECTION",
+    "//": "This should be mainly bad effects, since your unstable grasp on reality will further weaken when the reality stabilizes in lack of portal storms.",
+    "recurrence_min": "15 minutes",
+    "recurrence_max": "120 minutes",
+    "condition": {
+      "and": [
+        { "u_has_trait": "PORTAL_DEPENDENT" },
+        { "not": { "is_weather": "portal_storm" } },
+        { "not": { "u_has_worn_with_flag": "DIMENSIONAL_ANCHOR" } }
+      ]
+    },
+    "deactivate_condition": {
+      "or": [
+        { "not": { "u_has_trait": "PORTAL_DEPENDENT" } },
+        { "is_weather": "portal_storm" },
+        { "u_has_worn_with_flag": "DIMENSIONAL_ANCHOR" }
+      ]
+    },
```

```suggestion
    "deactivate_condition": {
      "or": [
        { "not": { "u_has_trait": "PORTAL_DEPENDENT" } },
        { "is_weather": "portal_storm" }
      ]
    },
```
For deactivation conditions you only want references to traits,bionics or weather.  Since deactivated conditions only get refreshed when those things change.

---

Done! Thanks for letting me now!